src/common/buffer.cc

```diff
@@ -1491,14 +1821,15 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
     }
   }
 
+  /* reverse previous logic. Now if any buffers can zero-copy then return true */
```

Comment:
why shall we reverse the previous logic? is there a performance gain if we do `for (ptr : buffers) {ptr->copy_to_fd(fd);}` instead of `writev()` if there are any buffer does not support zero copy?